diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/files/FileStructureRule.kt

```diff
@@ -123,19 +124,31 @@ class FileStructureRule(private val configRules: List<RulesConfig>) : Rule("file
                 // taking nodes with actual code
                 !it.isWhiteSpace() && !it.isPartOfComment() &&
                         // but not the ones we are going to move
-                        it.elementType != FILE_ANNOTATION_LIST && it.elementType != IMPORT_LIST &&
-                        // if we are here, then package is default and we don't need to select the empty PACKAGE_DIRECTIVE node
+                        it.elementType != FILE_ANNOTATION_LIST &&
+                        // if we are here, then IMPORT_LIST either is not present in the AST, or is empty. Either way, we don't need to select it.
+                        it.elementType != IMPORT_LIST &&
+                        // if we are here, then package is default and we don't need to select the empty PACKAGE_DIRECTIVE node.
                         it.elementType != PACKAGE_DIRECTIVE
             }
             ?: return  // at this point it means the file contains only comments
-        // fixme: handle other elements that could be present before package (other comments)
+        // We consider the first block comment of the file to be the one that possibly contains copyright information.
         var copyrightComment = firstCodeNode.prevSibling { it.elementType == BLOCK_COMMENT }
         var headerKdoc = firstCodeNode.prevSibling { it.elementType == KDOC }
+        // Annotations with target`file` can only be placed before `package` directive.
         var fileAnnotations = node.findChildByType(FILE_ANNOTATION_LIST)
+        // We also collect all other elements that are placed on top of the file.
+        // These may be other comments, so we just place them before the code starts.
+        val otherNodesBeforeCode = firstCodeNode.siblings(forward = false)
+            .filterNot {
+                it.isWhiteSpace() ||
+                        it == copyrightComment || it == headerKdoc || it == fileAnnotations
```

Comment:
What about EOL comments? Do we need to handle it here?

Reply:
We state the following structure (as per our code style): copyright information is placed in a block comment, general file comment is placed in KDoc. All other nodes that can appear before code, including EOL comments, are captured here.